### PR TITLE
ModalAlert: Add documentation and best practices

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Add documentation and examples  for ModalAlert

#### Why?

Add Best Practice and Variant sandpack examples.
Also updates the padding on Modal and ModalAlert footers

### Links

- [TDD](https://paper.dropbox.com/doc/AlertModal--Brestwoos1OliDMOFV~zn7U0Ag-tVQfnyLc7ZOTFmabB5PVt)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/docs-components/COMPONENT_DATA.js [hunk 1]
```diff
@@ -1390,6 +1390,56 @@ const GENERAL_COMPONENT_LIST: $ReadOnlyArray<ListItemType> = [
       figma: null,
     },
   },
+  {
+    svg: <Modal />,
```
Comment (ayeshakmaz) [inline docs/docs-components/COMPONENT_DATA.js:1394]: Using the same svg for now, but will update once I make a new illustration

File: docs/pages/web/modalalert.js [hunk 2]
```diff
@@ -79,64 +70,143 @@ export default function ModalAlertPage({ generatedDocGen }: {| generatedDocGen:
...
+              />
+            }
           />
           <MainSection.Card
-            cardSize="md"
-            type="don't"
-            description="Use language that makes it hard to understand what action is being taken, while adding additional actions that may take the user out of their existing context."
+            cardSize="lg"
+            type="do"
+            description="Use to overlay Page content. ModalAlerts should be horizontally and vertically centered on the screen."
```
Comment (AlbertCarreras) [inline docs/pages/web/modalalert.js:90]: Is this one required? I dont think this is specific for ModalAlert but for all Modals?

File: docs/examples/modalalert/doProvideAction.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  return (
+    <Layer zIndex={zIndex}>
+      <ModalAlert
+        type="error"
+        accessibilityModalLabel="Unable to charge credit card"
+        heading="Your credit card has expired"
+        primaryAction={{
+          accessibilityLabel: 'Update credit card info',
+          label: 'Update credit card',
+          href: 'https://www.pinterest.com',
```
Comment (AlbertCarreras) [inline docs/examples/modalalert/doProvideAction.js:18]: Are we adding the visit icon to all primary actions with URLs? What;s the intention of the icon?

**Review by AlbertCarreras — APPROVED** [review]

File: docs/examples/modalalert/doProvideAction.js (hunk 3 above)
Comment (ayeshakmaz) [reply docs/examples/modalalert/doProvideAction.js:18]: The icon showcases that folks are going to be navigated away from the modal to a new page that isn't related to the page they're currently on

File: docs/pages/web/modalalert.js (hunk 2 above)
Comment (ayeshakmaz) [reply docs/pages/web/modalalert.js]: Good call, removed

**Pull request merged**